Turn mutt(1) into a non-interactive program.

$ export AFL_STDIN=$(printf '\7\nhqq')
$ afl-fuzz [options] -- /path/to/mutt -R -f @@

--- a/init.c
+++ b/init.c
@@ -3491,6 +3491,7 @@
     Spoolfile = safe_strdup (buffer);
   }
 
+#ifdef __AFL_COMPILER
   if ((p = getenv ("MAILCAPS")))
     MailcapPath = safe_strdup (p);
   else
@@ -3498,6 +3499,7 @@
     /* Default search path from RFC1524 */
     MailcapPath = safe_strdup ("~/.mailcap:" PKGDATADIR "/mailcap:" SYSCONFDIR "/mailcap:/etc/mailcap:/usr/etc/mailcap:/usr/local/etc/mailcap");
   }
+#endif
 
   Tempdir = safe_strdup ((p = getenv ("TMPDIR")) ? p : "/tmp");
 
--- a/main.c
+++ b/main.c
@@ -797,12 +797,14 @@
       exit (0);
   }
 
+#ifndef __AFL_COMPILER
   /* Check for a batch send. */
   if (!isatty (0) || queries || alias_queries || dump_variables)
   {
     set_option (OPTNOCURSES);
     sendflags = SENDBATCH;
   }
+#endif
 
   /* Always create the mutt_windows because batch mode has some shared code
    * paths that end up referencing them. */
@@ -881,6 +883,7 @@
     /* we're not connected yet - skip mail folder creation */
     if (!mx_is_imap (fpath))
 #endif
+#ifndef __AFL_COMPILER
     if (stat (fpath, &sb) == -1 && errno == ENOENT)
     {
       snprintf (msg, sizeof (msg), _("%s does not exist. Create it?"), Maildir);
@@ -890,8 +893,28 @@
 	  mutt_error ( _("Can't create %s: %s."), Maildir, strerror (errno));
       }
     }
+#endif
   }
-
+#ifdef __AFL_HAVE_MANUAL_CONTROL
+   __AFL_INIT();
+#endif
+#ifdef __AFL_COMPILER
+  if (!isatty(0))
+  {
+    int pipefd[2];
+    if (pipe(pipefd))
+      abort();
+    const char *input = getenv("AFL_STDIN");
+    if (input == NULL)
+      abort();
+    if (dprintf(pipefd[1], "%s", input) != strlen(input))
+      abort();
+    if (dup2(pipefd[0], 0))
+      abort();
+    close(pipefd[0]);
+    close(pipefd[1]);
+  }
+#endif
   if (sendflags & SENDPOSTPONED)
   {
     if (!option (OPTNOCURSES))
--- a/mx.c
+++ b/mx.c
@@ -21,6 +21,10 @@
 # include "config.h"
 #endif
 
+#ifdef __AFL_COMPILER
+#undef USE_DOTLOCK
+#endif
+
 #include "mutt.h"
 #include "mx.h"
 #include "rfc2047.h"
